src/AutoRest.CSharp/MgmtTest/AutoRest/MgmtTestTarget.cs

```diff
@@ -23,25 +28,112 @@ public static void Execute(GeneratedCodeWorkspace project, CodeModel codeModel,
+                extensionWrapperTestWriter.Write();
+
+                project.AddGeneratedFile($"Mock/{extensionWrapperTest.Type.Name}.cs", extensionWrapperTestWriter.ToString());
+            }
+            catch
+            {
+            }
 
-            project.AddGeneratedFile($"Mock/{extensionWrapperTest.Type.Name}.cs", extensionWrapperTestWriter.ToString());
+
+            if (Configuration.MgmtConfiguration.TestModeler.GenerateSdkSample == false)
+                return;
+
+            var generated = new HashSet<string>();
+            foreach (var collectionTest in library.ResourceCollectionMockTests)
+            {
+                foreach (var testCase in collectionTest.MockTestCases)
+                {
+                    try
+                    {
+                        var collectionTestWriter = new ResourceCollectionMockTestWriter(new CodeWriter(), collectionTest);
+                        collectionTestWriter.WriteSample(testCase);
+                        var filename = GenExampleFileName(testCase);
+                        if (!generated.Contains(filename))
+                        {
+                            project.AddGeneratedFile(filename, collectionTestWriter.ToString());
+                            generated.Add(filename);
+                        }
+                    }
+                    catch
+                    {
+                    }
+                }
+            }
+
+            foreach (var resourceTest in library.ResourceMockTests)
+            {
+                foreach (var testCase in resourceTest.MockTestCases)
+                {
+                    try
+                    {
+                        var resourceTestWriter = new ResourceMockTestWriter(new CodeWriter(), resourceTest);
+                        resourceTestWriter.WriteSample(testCase);
+                        var filename = GenExampleFileName(testCase);
+                        if (!generated.Contains(filename))
+                        {
+                            project.AddGeneratedFile(filename, resourceTestWriter.ToString());
+                            generated.Add(filename);
+                        }
+                    }
+                    catch
+                    {
+                    }
+                }
+            }
+            foreach (var testCase in extensionWrapperTest.MockTestCases)
+            {
+                try
+                {
+                    var extensionWrapperTestWriter2 = new ExtensionWrapMockTestWriter(new CodeWriter(), extensionWrapperTest, library.ExtensionMockTests);
+                    extensionWrapperTestWriter2.WriteSample(testCase);
+                    var filename = GenExampleFileName(testCase);
+                    if (!generated.Contains(filename))
+                    {
+                        project.AddGeneratedFile(GenExampleFileName(testCase), extensionWrapperTestWriter2.ToString());
+                        generated.Add(filename);
```

Comment:
There here if we have the example generated with the same name, we just skip the current one. Will this make us to miss some examples?
How about we append some indices after the filename or something?